skimage/util/dtype.py

```diff
@@ -125,7 +125,10 @@ def _scale(a, n, m, copy=True):
         # Numbers can be represented exactly only if m is a multiple of n
         # Output array is of same kind as input.
         kind = a.dtype.kind
-        if n == m:
+        if n > m and a.max() < 2 ** m:
+            warn("Downcasting directly without scaling")
```

Comment:
I like this a lot, thanks @blink1073!

Is it easy to add the type and value info? e.g. "Downcasting uint64 to uint8 without scaling because max value 237 fits in uint8".